libnmstate/schema.py

```diff
@@ -437,3 +439,14 @@ class Mode:
 class MacVtap(MacVlan):
     TYPE = InterfaceType.MAC_VTAP
     CONFIG_SUBTREE = "mac-vtap"
+
+class Wireguard:
+    TYPE = InterfaceType.WIREGUARD
+    CONFIG_SUBTREE = "wireguard"
+    FWMARK = "fwmark" 
+    IPV4_AUTO_DEFAULT_ROUTE = "ipv4-auto-default-route"
+    IPV6_AUTO_DEFAULT_ROUTE = "ipv6-auto-default-route"	 
+    LISTEN_PORT = "listen-port" 	 	 
+    MTU = "mtu" 	 	 
+    PEER_ROUTES = "peer-routes" 	 	 
```

Comment:
The peer-routes can be done by existing route support of nmstate. No need to depend NM design for this.